marklogic-data-hub/src/main/java/com/marklogic/hub/dhs/installer/deploy/DeployHubAmpsCommand.java

```diff
@@ -0,0 +1,68 @@
+package com.marklogic.hub.dhs.installer.deploy;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.appdeployer.command.CommandContext;
+import com.marklogic.appdeployer.command.security.DeployAmpsCommand;
+import com.marklogic.mgmt.ManageClient;
+import com.marklogic.mgmt.SaveReceipt;
+import com.marklogic.mgmt.resource.ResourceManager;
+import com.marklogic.mgmt.resource.security.AmpManager;
+import com.marklogic.mgmt.util.ObjectMapperFactory;
+import org.springframework.web.client.HttpClientErrorException;
+
+/**
+ * This command is used instead of DeployAmpsCommand when a data-hub-security-admin user is deploying
+ * amps. It's because PUT request fail due to https://bugtrack.marklogic.com/55309.
+ */
+public class DeployHubAmpsCommand extends DeployAmpsCommand {
+
+    @Override
+    public boolean cmaShouldBeUsed(CommandContext context) {
+        return false;
+    }
+
+    @Override
+    protected SaveReceipt saveResource(ResourceManager mgr, CommandContext context, String payload) {
+        SaveReceipt receipt = null;
+        ManageClient manageClient = context.getManageClient();
+        try {
+            receipt = super.saveResource(mgr, context, payload);
+        }
+        catch (HttpClientErrorException ex) {
+            try{
+                JsonNode error = ObjectMapperFactory.getObjectMapper().readTree(ex.getResponseBodyAsString());
+                if (error.has("errorResponse")) {
+                    JsonNode errorResponse = error.get("errorResponse");
+                    if (errorResponse.has("messageCode")) {
+                        final String statusCode = errorResponse.get("statusCode").asText();
+                        final String status = errorResponse.get("status").asText();
+                        if ("Forbidden".equalsIgnoreCase(status) && "403".equals(statusCode)) {
+                            boolean ampExists;
+                            try{
+                                ampExists  = new AmpManager(manageClient).ampExists(payload);
+                            }
+                            catch (Exception e){
+                                throw ex;
+                            }
+                            if(ampExists){
+                                logger.error("Updates to amps are not allowed in this version of MarkLogic server.");
```

Comment:
There's a chance that a user will run this as an admin, so the message should be more informative here - e.g. "An error occurred while trying to update an amp: " + ex.getMessage() + ". Updates to amps as a user with the data-hub-security-admin role are not allowed in this version of MarkLogic, so this error is being logged and not thrown. If you need to modify the amp, you must delete it first via the Manage API."